src/commands.cpp

```diff
@@ -394,18 +390,199 @@ void AOClient::cmdFlip(int argc, QStringList argv)
     sendServerMessage(sender_name + " flipped a coin and got " + face + ".");
 }
 
+void AOClient::cmdDoc(int argc, QStringList argv)
+{
+    QString sender_name = ooc_name;
+    AreaData* area = server->areas[current_area];
+    if (argc == 0) {
+        sendServerMessage("Document: " + area->document);
+    }
+    else {
+        area->document = argv.join(" ");
+        sendServerMessage(sender_name + " changed the document."); // broadcast this!
```

Comment:
can you cherry pick the implementation of the function you want to use, and then update this branch to use it properly